addons/assigngear/functions/fn_replacePrimaryWeapon.sqf

```diff
@@ -46,7 +46,14 @@ if(count _this > 3) then {
         if (!isNil "_x" && _x != "" ) then {
             _unit addPrimaryWeaponItem _x;
         };
-    } forEach [_scope,_bipod,_attachment,_silencer];;
+    } forEach [_scope,_bipod,_attachment,_silencer,_magazine];
+
+    if(isNil "_magazineArray") then {_magazineArray = []};
```

Comment:
Probably better to edit params to be
` params ["_unit","_weaponArray","_scopeArray","_bipodArray","_attachmentArray","_silencerArray", ["_magazineArray",[]]];`

Infact, that goes for all of the parameters there